main/tests/test_main.cpp

```diff
@@ -69,8 +71,20 @@ const char **tests_get_names() {
 }
 
 MainLoop *test_main(String p_test, const List<String> &p_args) {
+	doctest::Context test_context;
+
+	// tests should be ordered by name for humans
+	test_context.setOption("order-by", "name");
+	test_context.setOption("abort-after", 5);
+	test_context.setOption("no-breaks", true);
+	test_context.run(); // run tests
 
-	if (p_test == "string") {
+	if (test_context.shouldExit())
+		return NULL;
+
+	return NULL;
+
+	/*if (p_test == "string") {
```

Comment:
I'm not sure what the project maintainers' stances are on this, but personally, I find that including commented out code into commits is... "unclean." After all, we have a version control system to see file history and revert things. I would suggest removing all commented out code from this PR.

Reply:
As far as I understand it, yes, the godot team typically doesn't want commented-out code.

That said, this PR still looks like it's a work-in-progress, so I wouldn't worry about the commented-out code just yet. When it gets to the point that this PR is considered finished and ready for review, then we can concern ourselves with clean code. :)

Reply:
Yes, this is most definitely a WIP.

Hence the commented out code. DW rebase is my friend. :)

Reply:
Resolved, closing. 

Rebase will happen at end.